lib/OktaAuth.ts

```diff
@@ -244,19 +245,17 @@ class OktaAuth implements SigninAPI, SignoutAPI {
     return isInteractionRequiredError(error);
   }
 
-  /**
-   * Alias method of signInWithCredentials
-   * 
-   * @todo This method is deprecated. Remove it in 5.0
-   */ 
-  signIn(opts) {
-    if (this.features.isLocalhost()) {
-      deprecate('This method has been deprecated, please use signInWithCredentials() instead.');
+  async signIn(opts: SigninOptions = { useInteractionCodeFlow: true }): Promise<AuthTransaction> {
```

Comment:
options passed from the devs will override the `useInteractionCodeFlow` to false if it's not explicitly added in the options object. 

Probably can set the default value at runtime when destructing the object:
```
const { useInteractionFlow = true, ...rest } =  SigninOptions;
```

Reply:
The intent is that if you call this method with no arguments, it will do an interaction code flow (this is a new feature). However if you pass `{ username, password }` it will do Authn V1 (compatibility) by default, as it does now. Passing `{username, password, useInteractionCodeFlow}` will use IDX and auto-fill remediations.